govcd/certificate_management.go

```diff
@@ -295,3 +326,36 @@ func (certificate *Certificate) Delete() error {
 
 	return nil
 }
+
+// getCertificateText returns the stripped text of the certificate, without the
+// starting and ending markers
+func getCertificateText(pem string) (string, error) {
+	reText, err := regexp.Compile(
+		`(?s)` + // treats newline as any other character
+			`-----BEGIN CERTIFICATE-----` + // the 'begin certificate' marker
+			`(.+)` + // any sequence of characters after the 'begin certificate' marker
+			`-----END CERTIFICATE-----`) // the 'end certificate' marker
+	if err != nil {
+		return "", err
+	}
+
+	text := reText.FindStringSubmatch(pem)
+	if len(text) < 2 {
+		return "", fmt.Errorf("start marker 'BEGIN CERTIFICATE' or end marker 'END CERTIFICATE' not found in input certificate")
+	}
+	return text[1], nil
+}
+
+// Same returns true if the certificate text matches the text of the provided PEM
+// (without the BEGIN CERTIFICATE and END CERTIFICATE markers)
+func (certificate *Certificate) Same(pem string) (bool, error) {
```

Comment:
For you consideration:

```suggestion
func (certificate *Certificate) IsSame(pem string) (bool, error) {
```

Reply:
I like `SameAs` - reads well

Reply:
Sounds great!

Reply:
Changed.